api/storage_service.cc

```diff
@@ -77,7 +77,7 @@ static ss::token_range token_range_endpoints_to_json(const dht::token_range_endp
     r.rpc_endpoints = d._rpc_endpoints;
     for (auto det : d._endpoint_details) {
         ss::endpoint_detail ed;
-        ed.host = det._host;
+        ed.host = boost::lexical_cast<std::string>(det._host);
```

Comment:
lexical_cast is too terrible for sstring->std::string

Reply:
Ah, I see you changed it to inet_address. Well, there is to_sstring() or just fmt::format().

Reply:
not part of this PR, explained below